### ISSUE TITLE
Error : Handshake status 401 Unauthorized

### ISSUE THREAD

**Issue by vmajor** [5ymph0en1x/SyBrain#5]
I am trying to evaluate this script on the BitMEX test account, but I cannot get past the first hurdle - logging in.

I am using the correct key and secret generated on testnet and I set line 31 to 'Test' but I get the 401 error.

Any advice? Is this caused by BitMEX or a change in their API?

**Comment by 5ymph0en1x** [comment]
Try putting `wss://testnet.bitmex.com/realtime` at line 11...

**Comment by vmajor** [comment]
Thanks for that. I could have sworn I tried it before but got a 404. Must have made a typo. 

Now I get passed the unauthorised and onto the error already reported by others:

`/usr/local/lib/python3.6/dist-packages/swagger_spec_validator/validator20.py:53: SwaggerValidationWarning: Found "$ref: #/definitions/UserPreferences" with siblings that will be overwritten. See https://stackoverflow.com/a/48114924 for more information. (path #/definitions/User/properties/preferences)
  ref_dict['$ref'], '/'.join(path),
It began in Africa...
Launching Machine learning Module...
Start: 2020-01-02 03:45:00 / End: 2020-01-02 03:50:00
time data '2020-01-02 03:45:00+00:00' does not match format '%Y-%m-%d %H:%M:%S%z'
 This is the end !
Traceback (most recent call last):
  File "sybrain.py", line 367, in <module>
    bmex()
  File "sybrain.py", line 276, in bmex
    while ws_bmex.ws.sock.connected:
AttributeError: 'NoneType' object has no attribute 'connected'`

It appears that time data is appending GMT offset to the GMT time, which is not helpful and it breaks the expected format. Any suggestion with this?

**Comment by 5ymph0en1x** [comment]
Ok. I have an ugly quick fix for linux.
In stockpredictor.py, line 91, put this instead : 
`D[i] = datetime.strptime(j, '%Y-%m-%d %H:%M:%S+00:00').replace(tzinfo=None).timestamp()`
It should work...

PS : Did you get my email ?

**Comment by vmajor** [comment]
That works, and yes I did. I just did not want to disturb you too much :)

Presently I am running SyBrain and SyTrage. I am not sure what SyTrage does as all it gives me are a list of current price directions for various currency pairs, or it exits with an error or silently. There may be something wrong if instead of exiting it was supposed to place an order, and did not. I will let you know once I figure out what may be going on.

I am however really interested in getting the Heptet to work, but that may require extra effort to make it Linux compatible so I am not going to push you...
